.github/workflows/ci.yml

```diff
@@ -39,4 +37,169 @@ jobs:
+            
+  ksat-demo-job: 
+    name: Ksat Job
+    runs-on: ${{ matrix.os }}
+    strategy: 
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+      with: 
+        python-version: '3.9'
+    - uses : conda-incubator/setup-miniconda@v3
+      with:
+        auto-update-conda: true
+        activate-environment: impalib
+        python-version: '3.9'
+    - name: CPP Header-Only Code
+      shell: bash -el {0}
+      run: |
+            conda activate impalib
+            python3 -m pip install --user .
+            cd examples/Ksat
+            cmake -B build && cmake --build build && cd build && ./demo
+    - name: Pure Python Code
+      shell: bash -el {0}
+      run: |
+            conda activate impalib
+            cd test/python_ksat/src
+            python3 main_ksat.py --filteringFlag=True --threshold=-0.0001 --nITER=200 --alpha=0.5 --randomTestFlag=True --nConstraints=20 --nVariables=10 --kVariable=3
+    - name: Python (CPP) Wrapper Code
+      shell: bash -el {0}
+      run: |
+            conda activate impalib
+            cd src/impa
+            python3 main_ksat.py --filteringFlag=True --threshold=-0.0001 --nITER=200 --inputPath=inputs_ksat_zeromean_var_10_samples_500 --outputPath=outputs_ksat_zeromean_var_10_samples_500 --PPElements=2 --testFile=112 --alpha=0.5 --var=8 --typeMetrics=2 --overwrite=1 --PPFlag=True
+            python3 main_ksat.py --filteringFlag=True --threshold=-0.0001 --nITER=200 --PPElements=2 --alpha=0.5 --var=8 --randomTestFlag=True --nConstraints=200 --nVariables=90 --kVariable=5 --PPFlag=True
+
+  kcmwm-demo-job:
+    name: Kcmwm Job
+    runs-on: ${{ matrix.os }}
+    strategy: 
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+      with: 
+        python-version: '3.9'
+    - uses: conda-incubator/setup-miniconda@v3
+      with:
+        auto-update-conda: true
+        activate-environment: impalib
+        python-version: '3.9'
+    - name: CPP Header-Only Code
+      shell: bash -el {0}
+      run: |
+            conda activate impalib
+            python3 -m pip install --user .
+            cd examples/KcMwm
+            cmake -B build && cmake --build build && cd build && ./demo
+            conda deactivate
+    - name: Pure Python Code
+      shell: bash -el {0}
+      run: |
+            conda activate impalib
+            cd test/python_kc_mwm/src
+            python3 main_pure_optimized.py --nITER=400 --filteringFlag=True --alpha=0.9 --PPFlag=True --threshold=-0.0001
+    - name: Python (CPP) Wrapper Code
+      shell: bash -el {0}
+      run: |
+            conda activate impalib
+            cd src/impa
+            python3 main_kc_mwm.py --nITER=400 --filteringFlag=True --alpha=0.9 --PPFlag=True --PPOption=1 --threshold=-0.0001
+  
```

Comment:
I could not verify the usage of ".txt" files as this is on GitHub, but was wondering if we need to delete the ".txt" files when unit tests are done

Reply:
The CI pipeline is currently checking these files to know if it needs to fail based on the contents of each file. If we delete the file inside of these `test_utils.py` files then we won't be able to make use of what the pipeline is trying to do. We can delete the file in the pipeline itself, however I've updated all the `with open ("example.txt", "a") as file:` to `with open ("example.txt", "w") as file:` that way the file is simply overwritten with the latest run's contents, not appended. Let me know what you think